src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,200 @@
+        while (iterator.hasNext()) {
+            add(iterator.next());
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexBounds(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexBounds(index);
+        Node<T> node = findNodeByIndex(index);
+        T oldItem = node.item;
+        node.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexBounds(index);
+        if (index == size - 1) {
+            return removeLast();
+        }
+        if (index == 0) {
+            return removeFirst();
+        }
+        Node<T> nodeToRemove = findNodeByIndex(index);
+        unlink(nodeToRemove);
+        return nodeToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if (current.item == object || current.item != null
+                    && current.item.equals(object)) {
+                unlink(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> prev;
+        private Node<E> next;
+
+        public Node(Node<E> prev, E item, Node<E> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private void addAtTheBeginning(T value) {
+        Node<T> first = head;
+        Node<T> newNode = new Node<>(null, value, first);
+        head = newNode;
+        if (first == null) {
+            tail = newNode;
+        } else {
+            first.prev = newNode;
+        }
+        size++;
+    }
+
+    private void addBeforeNode(T value, Node<T> node) {
+        Node<T> prevNode = node.prev;
+        Node<T> newNode = new Node<>(prevNode, value, node);
+        node.prev = newNode;
+        prevNode.next = newNode;
+        size++;
+    }
+
+    private T removeLast() {
+        final T element = tail.item;
+        Node<T> prevLast = tail.prev;
+        tail = prevLast;
+        if (prevLast == null) {
+            head = null;
+        } else {
+            prevLast.next = null;
+        }
+        size--;
+        return element;
+    }
+
+    private T removeFirst() {
+        final T element = head.item;
+        Node<T> nextFirst = head.next;
+        head = nextFirst;
+        if (nextFirst == null) {
+            tail = null;
+        } else {
+            nextFirst.prev = null;
+        }
+        size--;
+        return element;
+    }
```

Comment:
Delete these methods if you decide to not use them here: 
https://github.com/mate-academy/jv-linked-list/pull/904/files#r1114608108